theano/tensor/nnet/abstract_conv.py

```diff
@@ -2247,11 +2228,12 @@ def perform(self, node, inp, out_):
                                         # Deciding which kernel to select
                                         reg_row = row - ((kern.shape[4] - 1) - k_row)
                                         reg_col = col - ((kern.shape[5] - 1) - k_col)
-                                        reg_row = max(0, min(reg_row, kern.shape[2] - 1))
-                                        reg_col = max(0, min(reg_col, kern.shape[3] - 1))
+                                        if reg_row not in range(kern.shape[2]) or \
+                                                reg_col not in range(kern.shape[3]):
+                                            continue
```

Comment:
I don't think this is very efficient: it's creating these `range` objects for every pixel. Normal comparisons are probably faster. (And I'd invert the comparison, so you don't need the `continue`.)